src/main.py

```diff
@@ -168,19 +175,47 @@ def run_plan_bot(
             praw.models.Comment(reddit, _data=pushshift_comment)
         )
 
-        process_post(
-            comment,
-            plans,
-            posts_db,
-            post_ids_processed,
-            send=send_replies,
-            simulate=simulate_replies,
-            skip_tracking=skip_tracking,
-        )
+        process_the_post(comment)
+
+    # Get new comments since we last ran.
+    #
+    # subreddit.comments() returns the newest comments first so we
+    # need to keep track of that and save that as our cursor.  With no
+    # specified params, it returns newest 100 comments in the
+    # subreddit.
+    comments_params = get_comments_params(comments_progress_ref)
+
+    current_newest_comment_id = None
+    for comment in subreddit.comments(params=comments_params):
+        comment = reddit_util.Comment(comment)
+        if re.search("warrenplanbot", comment.text, re.IGNORECASE):
+            process_the_post(comment)
+
+        # update the cursor after processing the comment
+        if not current_newest_comment_id:
+            current_newest_comment_id = comment.fullname
+            if not skip_tracking:
+                comments_progress_ref.set(
+                    {"newest": current_newest_comment_id}, merge=True
```

Comment:
🤔 If comments are returned sorted from newest to oldest, won't this be progressively setting "newest" to older and older comments as it iterates through the list?

Reply:
I have a suggestion for how to fix this above :D

Reply:
It should only set the cursor the first time because of the `current_newest_comment_id` check unless I’m reading it wrong?

Reply:
Oh I see! You're right. I'd still propose to change the order from oldest to newest, and then to update this on each iteration. That'll have nice behavior for a mid-run crash :D